lib/engine/game/g_18_cz/step/sell_company_and_special_track.rb

```diff
@@ -38,6 +38,14 @@ def process_lay_tile(action)
                       else
                         @game.current_entity
                       end
+
+            unless @game.purple_tile?(action.tile)
+              discount = action.hex.tile.upgrades.sum(&:cost)
+              @log << "#{action.entity.owner.name} receives a discount of "\
+              "#{@game.format_currency(discount)} from "\
+              "#{action.entity.name}"
+            end
+
```

Comment:
This line in tracker.rb adds the discount info to the log. Is it not working for CZ?

https://github.com/crericha/18xx/blob/master/lib/engine/step/tracker.rb#L231

Reply:
I've just found something, a few lines up it's looking for a `:tile_discount` ability, but the abilities for these companies here in 18CZ are `:tile_lay` abilities so that's why it's not getting picked up by the existing code for logging the discounts.

The current tile laying code is kind of hard to follow everywhere with the different places costs get calculated and logged and everything, but improving that will take a pretty big effort, and is clearly beyond the scope of this PR 🙂